code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +626,237 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+	case XWMObjectType::oj_Mine4:
+		return "Defense Mine#Ion";
+	case XWMObjectType::oj_Satellite:
+		return "Sensor Satellite#Imp";
+	case XWMObjectType::oj_Nav_Buoy:
+		return "Nav Buoy#real";
+	case XWMObjectType::oj_Probe:
+		return "Sensor Probe";
+	case XWMObjectType::oj_Asteroid1:
+		return "Asteroid#Small01";
+	case XWMObjectType::oj_Asteroid2:
+		return "Asteroid#Small02";
+	case XWMObjectType::oj_Asteroid3:
+		return "Asteroid#Medium01";
+	case XWMObjectType::oj_Asteroid4:
+		return "Asteroid#Medium02";
+	case XWMObjectType::oj_Asteroid5:
+		return "Asteroid#Medium03";
+	case XWMObjectType::oj_Asteroid6:
+		return "Asteroid#Big01";
+	case XWMObjectType::oj_Asteroid7:
+		return "Asteroid#Big02";
+	case XWMObjectType::oj_Asteroid8:
+		return "Asteroid#Big03";
+	case XWMObjectType::oj_Training_Platform1:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform2:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform3:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform4:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform5:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform6:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform7:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform8:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform9:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform10:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform11:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform12:
+		return nullptr;
+	default:
+		break;
+	}
+	return nullptr;
+}
+
+vec3d xwi_determine_mine_formation_position(const XWMObject* oj, float objectPosX, float objectPosY, float objectPosZ, 
+	float offsetAxisA, float offsetAxisB)
+{
+	switch (oj->formation) {  // Y and Z axes must be switched for FSO
+	case XWMObjectFormation::ojf_FloorXY:
+		return vm_vec_new((objectPosX + offsetAxisA), objectPosZ, (objectPosY + offsetAxisB));
+	case XWMObjectFormation::ojf_SideYZ:
+		return vm_vec_new(objectPosX, (objectPosZ + offsetAxisA), (objectPosY + offsetAxisB));
+	case XWMObjectFormation::ojf_FrontXZ:
+		return vm_vec_new((objectPosX + offsetAxisA), (objectPosZ + offsetAxisB), objectPosY);
+	case XWMObjectFormation::ojf_Scattered:
+		return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+	default:
+		break;
+	}
+	return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+}
+
+void xwi_determine_object_orient(matrix* orient, const XWMObject* oj)
+{
+	angles a;
+	a.p = oj->object_pitch;
+	a.b = oj->object_roll;
+	a.h = oj->object_yaw;
+	vm_angles_2_matrix(orient, &a);
+	return;
+}
+
+void parse_xwi_objectgroup(mission* pm, const XWingMission* xwim, const XWMObject* oj)
+{
+	SCP_UNUSED(pm);
+	auto object_type = xwi_determine_object_type(oj);
+	if (object_type == nullptr)
+		return;
+
+	int number_of_objects = oj->numberOfObjects;
+	if (number_of_objects < 1)
+		return;
+
+	/**
+	NumberOfCraft Holds various parameter values depending on object types.For mines,
```

Comment:
I think NumberOfCraft is always 1 for space objects that aren't mines.  Add a check for this.  If the space object isn't a mine and has a number more than 1, display a `Warning` and set the number to 1.